src/js/fluxcontroller.js

```diff
@@ -511,6 +522,122 @@ define(function (require, exports, module) {
+
+                    if (!nextAction || (typeof nextAction !== "function")) {
+                        throw new Error("Exec passed an undefined action");
+                    }
+
+                    var params = Array.prototype.slice.call(arguments, 1),
+                        synchronizedAction = self._sychronizedActions.get(nextAction);
+
+                    return synchronizedAction.apply(null, params);
+                }
+            },
+
+            /**
+             * When the action queue and the JavaScript engine are idle, enqueue an action for execution.
+             *
+             * @param {string|Action} nextAction
+             * @return {Promise}
+             */
+            whenIdle: {
+                value: function () {
+                    var params = Array.prototype.slice.call(arguments, 0);
+
+                    var idleTaskPromise = new Promise(function (resolve, reject, onCancel) {
+                        var timer,
+                            idle;
+
+                        // When the queue becomes active, clear the timers and wait for it
+                        // to become idle again.
+                        var handleActive = function () {
+                            if (timer) {
+                                window.clearTimeout(timer);
+                                timer = null;
+                            }
+
+                            if (idle) {
+                                window.cancelIdleCallback(idle);
+                                idle = null;
+                            }
+
+                            // Wait for the queue to become idle again
+                            actionQueue.once("idle", handleIdle);
+                        };
+
+                        // When the queue become idle, wait for one second to ensure that it
+                        // stays idle, and then wait for the JavaScript engine to become idle
+                        // before finally enqueing the action and removing the queue-
+                        // activation listener.
+                        var handleIdle = function () {
+                            // Wait for the queue to quiesce for one second.
+                            timer = window.setTimeout(function () {
+                                // Wait for the JavaScript engine to become idle.
+                                idle = window.requestIdleCallback(function () {
+                                    // Clean up the queue-activation handler
+                                    actionQueue.off("active", handleActive);
+                                    timer = null;
+                                    idle = null;
+
+                                    // Enqueue the task
+                                    this.enqueue.apply(this, params)
+                                        .then(resolve, reject);
+                                }.bind(this));
+                            }.bind(this), 3000);
```

Comment:
Actually I was going to turn that down to 1 second, but I guess I forgot. There's no particular reason for that choice either though.